File: Client-side/Surveys.WebApplication/ClientApp/src/app/models/SurveyEntity.ts
import { SurveyDetail } from './SurveyDetail';

export class SurveyEntity {
  seid: number;
  title: string;
  descr: string;
  isOpen: boolean;
  createdBy: number;
  surveyDetails?: SurveyDetail;

  constructor(init?: Partial<SurveyEntity>) {
    Object.assign(this, init);
  }
  static ToListOfInstance(list: SurveyEntity[]): SurveyEntity[] {
    return list.map(item => { return new SurveyEntity(item); });
  }
  static ToInstance(item: SurveyEntity): SurveyEntity {
    return new SurveyEntity(item);
  }

}

File: Server-side/Surveys.BusinessLogic/DataAccess/SurveyEntity.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Runtime.Serialization;
using System.Text;

namespace Surveys.BusinessLogic.DataAccess
{
    [DataContract]
    public class SurveyEntity
    {
        [Key]
        [DataMember]
        public virtual int SEID { get; set; }

        [Required]
        [MaxLength(50)]
        [DataMember]
        public virtual string Title { get; set; }

        [DataMember]
        public virtual string Descr { get; set; }

        [DataMember]
        public virtual bool IsOpen { get; set; }

        [DataMember]
        public virtual int CreatedBy { get; set; }

        [DataMember]
        public virtual List<SurveyDetail> surveyDetails { get; set; }
    }

    public class Submitted
    {
        public virtual int SEID { get; set; }
    }
}
